project-base/storefront/components/Pages/Order/ContactInformation/utils.ts

```diff
@@ -0,0 +1,359 @@
+} from 'graphql/requests/orders/mutations/CreateOrderMutation.generated';
+import { TypeSimplePaymentFragment } from 'graphql/requests/payments/fragments/SimplePaymentFragment.generated';
+import { TypeListedStoreFragment } from 'graphql/requests/stores/fragments/ListedStoreFragment.generated';
+import { GtmMessageOriginType } from 'gtm/enums/GtmMessageOriginType';
+import { getGtmCreateOrderEventOrderPart, getGtmCreateOrderEventUserPart } from 'gtm/factories/getGtmCreateOrderEvent';
+import { onGtmCreateOrderEventHandler } from 'gtm/handlers/onGtmCreateOrderEventHandler';
+import { getGtmReviewConsents } from 'gtm/utils/getGtmReviewConsents';
+import { saveGtmCreateOrderEventInLocalStorage } from 'gtm/utils/gtmCreateOrderEventLocalStorage';
+import { Translate } from 'next-translate';
+import useTranslation from 'next-translate/useTranslation';
+import { NextRouter, useRouter } from 'next/router';
+import { OrderConfirmationQuery } from 'pages/order-confirmation';
+import { SubmitHandler, UseFormReturn, useWatch } from 'react-hook-form';
+import { ContactInformation } from 'store/slices/createContactInformationSlice';
+import { PageLoadingStateSlice } from 'store/slices/createPageLoadingStateSlice';
+import { usePersistStore } from 'store/usePersistStore';
+import { useSessionStore } from 'store/useSessionStore';
+import { CurrentCartType } from 'types/cart';
+import { CurrentCustomerType } from 'types/customer';
+import { OperationResult } from 'urql';
+import { ChangePaymentHandler, useChangePaymentInCart } from 'utils/cart/useChangePaymentInCart';
+import { useCurrentCart } from 'utils/cart/useCurrentCart';
+import { DomainConfigType } from 'utils/domain/domainConfig';
+import { handleFormErrors } from 'utils/forms/handleFormErrors';
+import { getIsPaymentWithPaymentGate } from 'utils/mappers/payment';
+import { getInternationalizedStaticUrls } from 'utils/staticUrls/getInternationalizedStaticUrls';
+import { useCurrentUserContactInformation } from 'utils/user/useCurrentUserContactInformation';
+
+type DeliveryInfo = Pick<
+    TypeCreateOrderMutationVariables,
+    | 'deliveryAddressUuid'
+    | 'deliveryCity'
+    | 'deliveryCompanyName'
+    | 'deliveryFirstName'
+    | 'deliveryLastName'
+    | 'deliveryPostcode'
+    | 'deliveryStreet'
+    | 'deliveryTelephone'
+    | 'isDeliveryAddressDifferentFromBilling'
+>;
+
+export const useContactInformationPageNavigation = () => {
+    const { url } = useDomainConfig();
+    const router = useRouter();
+    const [transportAndPaymentUrl] = getInternationalizedStaticUrls(['/order/transport-and-payment'], url);
+    const updatePageLoadingState = useSessionStore((s) => s.updatePageLoadingState);
+
+    const goToPreviousStepFromContactInformationPage = () => {
+        updatePageLoadingState({ isPageLoading: true, redirectPageType: 'order-process' });
+        router.push(transportAndPaymentUrl);
+    };
+
+    return { goToPreviousStepFromContactInformationPage };
+};
```

Comment:
I do not necessarily see the reason to complicate it here. Are you able to give some examples for which it would be necessary? We have a form submit handler there which can take care of it, but if you have some strong arguments, I am all ears.

Reply:
No problem, if there is any complication, let's keep it simple :)